### ISSUE TITLE
oce.plot.ts() does not put grid at right spots

### ISSUE THREAD

**Issue by dankelley** [dankelley/oce#2267]
See snapshot. Blue is from the `grid` arguments to `oce.plot.ts()` and green is from using the return value of that function with `abline()` calls, from code like the following. Notice that the green lines are where the axis has put the labels.  (I have checked, and the `g$xat` values are, in fact at the start of each month.)
```R
    g <- oce.plot.ts(t, y,grid = TRUE, grid.col = 4, grid.lty = 1, grid.lw = 1)
    abline(h=g$yat, col=3)
    abline(v=g$xat, col=3)
```

<img width="783" alt="Screenshot 2024-12-06 at 11 24 08 AM" src="https://github.com/user-attachments/assets/8f853a27-29db-4c8b-8dee-b764271c69c4">

**Comment by dankelley** [comment]
Here's a reprex:

``` r
library(oce)
#> Loading required package: gsw
t0 <- as.POSIXct("2024-01-01 00:00:00", tz = "UTC")
t <- seq(t0, by = "month", length.out = 6)
y <- seq_along(t)
g <- oce.plot.ts(t, y, grid = TRUE, grid.col = 4, grid.lty = 1)
abline(h = g$yat, col = 3)
abline(v = g$xat, col = 3)
```

![](https://i.imgur.com/B4JBKUN.png)

``` r
print(t)
#> [1] "2024-01-01 UTC" "2024-02-01 UTC" "2024-03-01 UTC" "2024-04-01 UTC"
#> [5] "2024-05-01 UTC" "2024-06-01 UTC"
print(g)
#> $xat
#> [1] "2024-01-01 UTC" "2024-02-01 UTC" "2024-03-01 UTC" "2024-04-01 UTC"
#> [5] "2024-05-01 UTC" "2024-06-01 UTC"
#> 
#> $yat
#> [1] 1 2 3 4 5 6
```

<sup>Created on 2024-12-06 with [reprex v2.1.1](https://reprex.tidyverse.org)</sup>

**Comment by richardsc** [comment]
Just searching through old issues, I found this:

https://github.com/dankelley/oce/issues/581

which reminded me of the `oce.grid()` function:

https://dankelley.github.io/oce/reference/oce.grid.html

Not sure if it's relevant, but figured I'd point it out.

**Comment by dankelley** [comment]
Hm, this seems to be caused by the use of the base function `axTicks()`.  I won't bother trying to write a paragraph to explain, since the snapshot shows it.  I will think about other ways, e.g. using `pretty()` to find where to put ticks.

<img width="1108" alt="Screenshot 2024-12-06 at 1 24 07 PM" src="https://github.com/user-attachments/assets/49c6e0f7-1f17-44da-b737-b41c1d8fb130">

**Comment by dankelley** [comment]
Yeah, `pretty()` may be the solution.
<img width="869" alt="Screenshot 2024-12-06 at 1 27 42 PM" src="https://github.com/user-attachments/assets/f7c8396e-2d79-46e2-bc20-dff0ea471344">

**Comment by dankelley** [comment]
Hm, `pretty()` does not always choose the labelled times.  I was sort of afraid of that.

I'll return to this on the weekend.

**Comment by dankelley** [comment]
I think `oce.axis.POSIXct()` is our friend here ... I'll try using that when I return to this.  When I instrument it, I indeed find times at which the labels have been drawn.


(Sorry for cryptic, image-based comments ... these are not meant for typical users.)

<img width="1222" alt="Screenshot 2024-12-06 at 2 40 07 PM" src="https://github.com/user-attachments/assets/6255effe-cdc9-4eb5-a965-e140a5f33a1f">

**Comment by dankelley** [comment]
I do think `oce.axis.POSIXct()` will be the solution.  It's working well in the (not-yet-pushed) test branch named issue2267.  I'll do more testing tomorrow, and merge into "develop" when it seems suitable.  (I may use it for a few days first.)

**Comment by dankelley** [comment]
This is solved in "develop" commit 6cbc81e2a94fdcbce3c33428834fb9f13cf3d34c and demonstrated in the R code at https://github.com/dankelley/oce-issues/tree/main/22xx/2267, which makes 48 plots.  See the README file in that directory for how to examine the plots to check the code.

NOTE: for now (maybe only for a week or so) the *old* vertical grid lines are shown (in blue) if you set `debug` to a positive integer. But that feature will go away, without comment, when I feel that it would be a distraction to users.
